auth/src/types/metadata.ts

```diff
@@ -0,0 +1,47 @@
+import { CountryCode, NSACodes } from "./countrycodes";
+
+/**
+ * User specific data
+ */
+export interface User {
+  id: string;
+  email: string;
+  email_verified: boolean;
+  family_name: string;
+  given_name: string;
+  locale: string;
+  pronouns: string;
+}
+
+export type Conference = number;
+export type Commitee = {
+  id: string;
+  conference: Conference;
+};
+
+export interface NonStateActor {
+  conference: Conference;
+  organization: NSACodes;
+}
+
+export interface Representative {
+  conference: Conference;
+  commitee: Commitee;
+  country: CountryCode;
+}
+
+export interface SecretaryMember {
+  conference: Conference;
+}
+
+export interface Chair {
+  commitee: Commitee;
+}
```

Comment:
Doesn't the chair needs a conference like the Delegate and NSA?

Reply:
These types only refer to the entities as the auth system sees them. Fields like the ones you described would typically live in a database and not be stored on the user. The types you find in this file are all stored in the user metadata which is used to authenticate a user and grant them permissions to do things. This data is included in the session token and therefore should be relatively slim

Reply:
Ah! But why are they needed in the delegates metadata?

Reply:
The metadata is what ZITADEL allows us to store per user so we can check for the permissions in the metadata on a per request level. Depending on what the user is, the metadata will include the permissions in this structure.